lxd/containers.go

```diff
@@ -54,22 +60,78 @@ func containersGet(d *Daemon, r *http.Request) Response {
 	}
 }
 
-func doContainersGet(d *Daemon) ([]string, error) {
+func doContainerGet(d *Daemon, cname string) (shared.ContainerInfo, Response) {
+	_, err := dbGetContainerId(d.db, cname)
+	if err != nil {
+		return shared.ContainerInfo{}, SmartError(err)
+	}
+
+	c, err := newLxdContainer(cname, d)
+	if err != nil {
+		return shared.ContainerInfo{}, SmartError(err)
+	}
+
+	var name string
+	regexp := fmt.Sprintf("%s/", cname)
+	length := len(regexp)
+	q := "SELECT name FROM containers WHERE type=? AND SUBSTR(name,1,?)=?"
```

Comment:
Nevermind, got confused for a while thinking you were querying containers instead of snapshots :)

Yes, your code is right for snapshots, sorry about that.